client/post-editor/editor-drawer/taxonomies.jsx

```diff
@@ -18,22 +18,34 @@ import Gridicon from 'components/gridicon';
 import TermTokenField from 'post-editor/term-token-field';
 import TermSelector from 'post-editor/editor-term-selector';
 
+function isSkippedTaxonomy( postType, taxonomy ) {
+	if ( includes( [ 'post_format', 'mentions' ], taxonomy ) ) {
+		return true;
+	}
+
+	if ( 'post' === postType ) {
+		return includes( [ 'category', 'post_tag' ], taxonomy );
+	}
+
+	return false;
+}
+
 function EditorDrawerTaxonomies( { siteId, postType, postTerms, taxonomies } ) {
 	return (
 		<div className="editor-drawer__taxonomies">
 			{ siteId && postType && (
 				<QueryTaxonomies { ...{ siteId, postType } } />
 			) }
-			{ map( taxonomies, ( taxonomy ) => {
+			{ reduce( taxonomies, ( memo, taxonomy ) => {
 				const { name, label, hierarchical } = taxonomy;
-				if ( 'post_format' === name ) {
-					// Post format has its own dedicated accordion
-					return;
+
+				if ( isSkippedTaxonomy( postType, name ) ) {
+					return memo;
 				}
 
 				const icon = hierarchical ? 'folder' : 'tag';
 
-				return (
+				return memo.concat(
 					<Accordion
 						key={ name }
 						title={ label }
```

Comment:
I am not sure how this got back in there but we can make this change after #6548:

```diff
- <TermTokenField postTerms={ postTerms } taxonomyName={ name } />
+ <TermTokenField taxonomyName={ name } />
```